libs/payment/api/src/lib/payrexx/gateway-client.ts

```diff
@@ -0,0 +1,73 @@
+import {PayrexxClient} from './payrexx-client'
+import {Transaction} from './transaction-client'
+
+type CreateGatewayRequestData = {
+  psp: number[]
+  pm: string[]
+  referenceId: string
+  amount: number
+  fields?: Record<string, any>
+  successRedirectUrl: string
+  failedRedirectUrl: string
+  cancelRedirectUrl: string
+  vatRate: number
+  currency: string
+  preAuthorization?: boolean
+  chargeOnAuthorization?: boolean
+}
+
+export type GatewayStatus = 'waiting' | 'confirmed' | 'cancelled' | 'declined'
+export type Gateway = {
+  id: number
+  status: GatewayStatus
+  hash: string
+  referenceId: string
+  link: string
+  invoices?: Array<{
+    transactions: Array<Transaction>
+  }>
+  preAuthorization: true
+  fields: []
+  psp: number[]
+  pm: string[]
+  amount: number
+  currency: string
+  vatRate: number
+  sku: string
+  applicationFee: number
+  createdAt: number
+}
+
+type CreateGatewayResponseData = Array<Gateway>
+type GetGatewayResponseData = Array<Gateway>
+
+export class GatewayClient {
+  constructor(private readonly client: PayrexxClient) {}
+
+  async createGateway(requestData: CreateGatewayRequestData) {
+    const response = await this.client.post<CreateGatewayResponseData>('Gateway', requestData)
+    if (response.status === 'error') {
+      throw new Error(`Payrexx request has error status with message: ${response.message}`)
```

Comment:
maybe we can specify that the error is about the payrexx gateway, not just payrexx in general